### PR TITLE
Update chat state notifications when message is edited.

### PR DESCRIPTION
**Pull request by legastero**
Chat state notifications for a sender are cleared out once a new message is added to the collection. For corrections, however, the existing message model is updated instead of a new one added.

This adds a `change:edited` hook to the messages collection so that the chat state notifications can be updated.

Resolves issue #1715

### CODE REVIEW

File: src/converse-chatview.js [hunk 1]
```diff
@@ -1072,6 +1082,16 @@ converse.plugins.add('converse-chatview', {
...
+            /**
+             * Remove chat state notifications for a given sender JID.
+             * @private
+             * @method _converse.ChatBoxView#clearChatStateForSender
+             * @param {string} sender - The sender of the chat state
+             */
+            clearChatStateForSender (sender) {
+                sizzle(`.chat-state-notification[data-csn="${sender}"]`, this.content).forEach(u.removeElement);
+            },
+
```
Comment (jcbrand) [inline src/converse-chatview.js]: Can you please also call `clearChatStateForSender` in `showMessage`?

File: src/converse-chatview.js [hunk 2]
```diff
@@ -196,6 +196,7 @@ converse.plugins.add('converse-chatview', {
                 this.initDebounced();
 
                 this.listenTo(this.model.messages, 'add', this.onMessageAdded);
+                this.listenTo(this.model.messages, 'change:edited', this.onMessageEdited);
```
Comment (jcbrand) [inline src/converse-chatview.js]: This is for one-on-one chats only, so we need a similar line in the `initialize` method of `_converse.ChatRoomView`.

File: src/converse-chatview.js (hunk 1 above)
Comment (legastero) [reply src/converse-chatview.js]: Already handled in this commit 😁

Unless there's another `showMessage` I need to search for? I see another one in the omemo view, but it invokes the one here.

Comment (jcbrand) [reply src/converse-chatview.js]: Sorry, I missed the relevant line.

**Comment by jcbrand** [comment]
Merged manually.

**Pull request closed without merging**